src/java.management/share/classes/javax/management/monitor/Monitor.java

```diff
@@ -1531,10 +1537,16 @@ public Void run() {
                     return null;
                 }
             };
-            if (ac == null) {
-                throw new SecurityException("AccessControlContext cannot be null");
+            if (!SharedSecrets.getJavaLangAccess().allowSecurityManager()) {
+                // No SecurityManager permitted:
+                Subject.doAs(s, action); // s is permitted to be null
```

Comment:
While `s` is permitted to be null, calling `Subject.doAs(null, action)` actually sets the current subject to null while calling `action`. This is not same as directly calling `action` where the current subject (could be non null) is used.

Reply:
Yes, good point, got it. 8-)